File: 554. Brick Wall.java
class Solution {
    public int leastBricks(List<List<Integer>> wall) {
        HashMap<Integer, Integer> sums = new HashMap<>();
        int rowSum = 0;
        for(List<Integer> layer : wall){
            int sum = 0;
            for(int brick : layer){
                sum += brick;
                sums.put(sum, sums.getOrDefault(sum, 0) + 1);
            }
            rowSum = sum;
        }
        int ans = wall.size();
        for(Map.Entry<Integer, Integer> ele : sums.entrySet()){
            if(ele.getKey() == rowSum) continue;
            ans = Math.min(ans, wall.size() - ele.getValue());
        }
        
        return ans;
    }
}

File: 554. Brick Wall.cpp
class Solution {
public:
    int leastBricks(const vector<vector<int>>& wall) {
        unordered_map<int, int> sums;
        long rowSum = 0;
        for (const auto& layer : wall) {
            long sum = 0;
            for (int brick : layer) {
                sum += brick;
                sums[sum]++;
            }
            rowSum = sum;
        }
        int ans = wall.size();
        for (const auto& ele : sums) {
            if (ele.first == rowSum) continue;
            ans = min(ans, static_cast<int>(wall.size()) - ele.second);
        }
        return ans;
    }
};

